ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/BlockValueCalculator.java

```diff
@@ -27,7 +27,13 @@ public Wei calculateBlockValue(final BlockWithReceipts blockWithReceipts) {
     Wei totalFee = Wei.ZERO;
     for (int i = 0; i < txs.size(); i++) {
       final Wei minerFee = txs.get(i).getEffectivePriorityFeePerGas(block.getHeader().getBaseFee());
-      totalFee = totalFee.add(minerFee.multiply(receipts.get(i).getCumulativeGasUsed()));
+      // we don't store gasUsed and need to calculate that on the fly
+      // receipts are fetched in ascendant sorted by cumulativeGasUsed
```

Comment:
```suggestion
      // we don't store gasUsed per transaction and need to calculate that on the fly
      // receipts are fetched in ascending sorted by cumulativeGasUsed
```

Reply:
Had the same thought, did a bit of digging into the code, and even the BlockchainQuery does this same calculation to create instances of TransactionReceiptWithMetadata